papergres.go

```diff
@@ -436,89 +447,90 @@ func (s *Schema) InsertAll(objs interface{}) ([]*Result, error) {
 		return nil, err
 	}
 	if len(slice) == 0 {
-		return nil, errors.New("Empty slice")
+		return nil, errors.New("empty slice")
 	}
 
-	// sql, args := insertMultipleSQL(objs, s.Name)
-	// return s.Database.Query(sql, args...).ExecNonQuery(), nil
-
 	// now turn the objs into a repeat query and exec
 	return s.GenerateInsert(slice[0]).Repeat(len(slice),
 		func(i int) (dest interface{}, args []interface{}) {
-			args = insertArgs(slice[i])
+			args = insertArgs(slice[i], false)
 			return
 		}).Exec()
 }
 
+// GenerateInsert generates the insert query for the given object
+func (s *Schema) GenerateInsert(obj interface{}) *Query {
+	return s.generateInsertQuery(obj, false)
+}
+
+// GenerateInsertWithPK generates the insert query for the given object in which
+// PrimaryKey value is also supposed to be populated during insert.
+func (s *Schema) GenerateInsertWithPK(obj interface{}) *Query {
+	return s.generateInsertQuery(obj, true)
+}
+
+// generateInsertQuery constructs an insert query for the given object
+func (s *Schema) generateInsertQuery(obj interface{}, withPK bool) *Query {
+	sql := insertSQL(obj, s.Name, withPK)
+	args := insertArgs(obj, withPK)
+	q := s.Database.Query(sql, args...)
+	q.insert = true
+	return q
+}
+
 // insertSQL generates insert SQL string for a given object and schema
-func insertSQL(obj interface{}, schema string) string {
+func insertSQL(obj interface{}, schema string, withPK bool) string {
+	// Construct the table name prefixed with schema name
 	tname := goToSQLName(getTypeName(obj))
 	tname = fmt.Sprintf("%s.%s", schema, tname)
+
+	// Construct the first component of insert statement
 	sql := fmt.Sprintf("INSERT INTO %s (", tname)
 
-	fields, primary := prepareFields(obj)
+	// NOTE: An object is represented as a slice of Fields, where each Field
```

Comment:
See above regarding comments.

Reply:
Same as above. Line comments are the norm.